### PR TITLE
[Sival, rv_timer] Test counter wrap 

### LINKED ISSUE #19978: [chip-test, SiVal] RV_TIMER Tick configuration
### Test point name

[tick_configuration_test](https://github.com/lowRISC/opentitan/blob/365b6f9c4144ecc8753c58290e186bf80fca0254/hw/top_earlgrey/data/ip/chip_rv_timer_testplan.hjson#L25C9-L25C9)

### Host side component

None

### OpenTitanTool infrastructure implemented

None

### Silicon Validation (SiVal)

Yes

### Emulation Targets

- [ ] None
- [x] CW310
- [x] Hyperdebug + CW310

### Contact person

matutem

### Checklist

Please fill out this checklist as items are completed. Link to PRs and issues as appropriate.
- [ ] Check if existing test covers most or all of this testpoint (if so, either extend said test to cover all points, or skip the next 3 checkboxes)
- [x] Device-side (C) component developed
- [x] Bazel build rules developed
- [x] ~Host-side component developed~
- [x] HJSON test plan updated with test name (so it shows up in the dashboard)
- [ ] Test added to dvsim nightly regression (and passing at time of checking)
- [ ] For SiVal test cases, test is running relevant FPGA or silicon regression

> estimate 2

### LINKED ISSUE #19979: [chip-test, sival] RV_TIMER riscv_isa_csrs
### Test point name

[riscv_isa_csrs](https://github.com/lowRISC/opentitan/blob/365b6f9c4144ecc8753c58290e186bf80fca0254/hw/top_earlgrey/data/ip/chip_rv_timer_testplan.hjson#L43C1-L43C1)

### Host side component

None Required

### OpenTitanTool infrastructure implemented

Yes

### Silicon Validation (SiVal)

Yes

### Emulation Targets

- [ ] None
- [X] CW310
- [X] Hyperdebug + CW310

### Contact person

matutem

### Checklist

Please fill out this checklist as items are completed. Link to PRs and issues as appropriate.
- [ ] Check if existing test covers most or all of this testpoint (if so, either extend said test to cover all points, or skip the next 3 checkboxes)
- [ ] Device-side (C) component developed
- [ ] Bazel build rules developed
- [ ] Host-side component developed
- [ ] HJSON test plan updated with test name (so it shows up in the dashboard)
- [ ] Test added to dvsim nightly regression (and passing at time of checking)
- [ ] For SiVal test cases, test is running relevant FPGA or silicon regression

> estimate 2

### PR DESCRIPTION
**Pull request by engdoreis**
The original test description does not make sense because the Ibex does not implement the timer CSRs as described [here](https://ibex-core.readthedocs.io/en/latest/03_reference/cs_registers.html?highlight=MTIME#machine-architecture-id-marchid) and therefore the rv_timer memory mapped registers are used instead.
This PR:
 - Updates the documentations to make it clear.
 - Update the test description to only check the counter wrap.
 - Update the existing systick_test exec_env.
 - Increment the systick_test to also check the counter wrap.

Fix https://github.com/lowRISC/opentitan/issues/19978
Fix https://github.com/lowRISC/opentitan/issues/19979

### CODE REVIEW

File: sw/device/tests/rv_timer_systick_test.c [hunk 1]
```diff
@@ -56,6 +57,37 @@ static status_t test_tick(uint32_t tick_hz) {
...
+
+  TRY(dif_rv_timer_counter_set_enabled(&timer, kHart, kDifToggleEnabled));
+  busy_spin_micros(kWrapTimeMillis * 1000 * 2);
+  TRY(dif_rv_timer_counter_set_enabled(&timer, kHart, kDifToggleDisabled));
+
+  uint64_t counter = UINT64_MAX;
+  TRY(dif_rv_timer_counter_read(&timer, kHart, &counter));
+  // Verify that the counter wrapped.
+  TRY_CHECK(counter < start_counter,
+            "Unexpected elapsed time, expected: %x%x, got: %x%x",
```
Comment (jwnrt) [inline sw/device/tests/rv_timer_systick_test.c:80]: Nit: Do these need to be `%x%08x` to make sense when joined together?

**Review by jwnrt — APPROVED** [review]
LGTM, thanks @engdoreis

**Review by pamaury — APPROVED** [review]
Looks good to me, I think James is right about the format missing some zero padding.

File: sw/device/tests/rv_timer_systick_test.c (hunk 1 above)
Comment (engdoreis) [reply sw/device/tests/rv_timer_systick_test.c:80]: This is the only to print 64 bits variables with the log macros.

Comment (engdoreis) [reply sw/device/tests/rv_timer_systick_test.c]: Good catch. Fixed.

**Pull request merged**